clippy_lints/src/vec.rs

```diff
@@ -132,9 +132,16 @@ impl<'tcx> LateLintPass<'tcx> for UselessVec {
     fn check_crate_post(&mut self, cx: &LateContext<'tcx>) {
         for (span, lint_opt) in &self.span_to_lint_map {
             if let Some((hir_id, suggest_slice, snippet, applicability)) = lint_opt {
-                let help_msg = format!("you can use {} directly", suggest_slice.desc(),);
+                let source_map = cx.tcx.sess.source_map();
+                let comments = span_extract_comments(source_map, *span);
```

Comment:
Since we aren't actually interested in the comment as a string, can this just use `span_contains_comment`?

Reply:
And yet another discovery, thanks!